### PR TITLE
Issue #105: Introducing `Seniority` entity to `Candidate` model

### LINKED ISSUE #105: Add `seniority` attribute to `Candidate` model
- [x] To update add command
- [x] To save as `COM1, COM2, COM3, COM4`
- [x] ... more coming

### PR DESCRIPTION
**Pull request by domlimm**
This PR consists of the following changes:
- Added `Seniority` entity to existing `Candidate` model
- Added `yr/[VALUE]` prefix for `Seniority
- Added test cases for `Seniority`
- Refactored files affected by this new entity

Closes #105.

### CODE REVIEW

**Comment by domlimm** [comment]
@lzan98 @tiewweijian hang tight guys! fixing the `codecov/patch` first.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -136,6 +137,29 @@ public static Course parseCourse(String course) throws ParseException {
...
+        try {
+            int seniorityInt = Integer.parseInt(trimmedSeniority.substring(trimmedSeniority.length() - 1));
+
+            if (!Seniority.isValidSeniority(seniorityInt)) {
+                throw new ParseException(Seniority.MESSAGE_CONSTRAINTS);
+            }
+
+            return new Seniority(seniorityInt);
+        } catch (NumberFormatException e) {
+            throw new ParseException(Seniority.MESSAGE_CONSTRAINTS);
```
Comment (tiewweijian) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:159]: Maybe we could write a test (in this PR, or next PR) to parse in an invalid number like `1.0` or `2.4` to induce this error so that this line is covered?

File: src/test/java/seedu/address/model/candidate/SeniorityTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        Integer invalidSeniority = null;
+        assertThrows(NullPointerException.class, () -> new Seniority(invalidSeniority));
+    }
+
+    @Test
+    public void constructor_invalidSeniority_throwsIllegalArgumentException() {
+        int invalidSeniority = 0;
+        assertThrows(IllegalArgumentException.class, () -> new Seniority(invalidSeniority));
+    }
+
```
Comment (tiewweijian) [inline src/test/java/seedu/address/model/candidate/SeniorityTest.java:21]: Was thinking we can write test cases here for more edge cases like if seniority is >= 1 and <= 4, but it's a `double` instead of an `int`

**Review by tiewweijian — APPROVED** [review]
Thanks for doing this up @domlimm ! Made some comments to add more test cases, but I think other than that it looks ok! (Not sure if you wanna include it in this PR, or the next one)

File: src/test/java/seedu/address/model/candidate/SeniorityTest.java (hunk 2 above)
Comment (domlimm) [reply src/test/java/seedu/address/model/candidate/SeniorityTest.java:21]: @tiewweijian thank you for reviewing and proposed suggestions have been added!

just a note, clarified certain things offline too.

**Review by lzan98 — APPROVED** [review]
LGTM! Great work 👍
